decompiler/config/jak2/all-types.gc

```diff
@@ -29690,21 +29693,36 @@
 ;; enemy-h                        ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-#|
-(deftype enemy-focus (focus)
+;;Added hopefully temp structure since "focus" used the 12 method slot instead of enemy-focus
+(deftype temp-focus (structure)
+  ((handle       handle        :offset-assert 0)
+   (collide-with collide-spec  :offset-assert 8)
+   )
+  :method-count-assert 12
+  :size-assert         #xc
+  :flag-assert         #xc0000000c
+  (:methods
+    (clear-focused (_type_) uint 9)
+    (collide-check? (_type_ process-focusable) object 10)
+    (reset-to-collide-spec (_type_ collide-spec) none 11)
+    )
+  )
+
+(declare-type enemy object)
+
+(deftype enemy-focus (temp-focus)
   ((aware uint64  :offset-assert 16)
    (flags uint64  :offset-assert 24)
    )
   :method-count-assert 14
   :size-assert         #x20
   :flag-assert         #xe00000020
   (:methods
-    (enemy-focus-method-13 () none 13)
+    (enemy-focus-method-12 (_type_ uint32 enemy) symbol 12)
+    (enemy-focus-method-13 (_type_ uint32 uint32) symbol 13)
```

Comment:
For arguments and return values, we should use `int` and `uint`, and not any of ones with sizes. The integer types with sizes should be used in arrays and fields of types (any place where they are stored in memory) and the ones without sizes should be used for any variable/argument/return value (any places where they are stored in registers)